testsuite/cluster-test/src/cluster_swarm/cluster_swarm_kube.rs

```diff
@@ -525,19 +530,34 @@ impl ClusterSwarmKube {
             .allocate_node(&pod_name)
             .await
             .map_err(|e| format_err!("Failed to allocate node: {}", e))?;
-        debug!("Configuring fluent-bit for pod {} on {:?}", pod_name, node);
+        debug!(
+            "Configuring fluent-bit, genesis and config for pod {} on {}",
+            pod_name, node.name
+        );
         match &instance_config.application_config {
-            Validator(_) => {
-                self.config_fluentbit("validator-events", pod_name.as_str(), &node.name)
-                    .await?
+            Validator(validator_config) => {
+                self.config_fluentbit("validator-events", &pod_name, &node.name)
+                    .await?;
+                if validator_config.enable_mgmt_tool {
+                    self.put_genesis_file(&pod_name, &node.name).await?;
+                    self.generate_config(&instance_config, &pod_name, &node.name)
+                        .await?;
+                }
             }
-            Fullnode(_) => {
-                self.config_fluentbit("validator-events", pod_name.as_str(), &node.name)
-                    .await?
+            Fullnode(fullnode_config) => {
+                self.config_fluentbit("validator-events", &pod_name, &node.name)
+                    .await?;
+                if fullnode_config.enable_mgmt_tool {
+                    self.put_genesis_file(&pod_name, &node.name).await?;
+                    self.generate_config(&instance_config, &pod_name, &node.name)
+                        .await?;
+                }
             }
             LSR(_) => {
-                self.config_fluentbit("safety-rules-events", pod_name.as_str(), &node.name)
-                    .await?
+                self.config_fluentbit("safety-rules-events", &pod_name, &node.name)
+                    .await?;
+                self.generate_config(&instance_config, &pod_name, &node.name)
```

Comment:
for LSR we always generate_config no matter ```enable_mgmt_tool```?

Reply:
enable_mgmt_tool is used to gate LSR/Vault. In the case of no LSR, this arm is not called.